src/main/java/no/unit/nva/cristin/institutions/model/Identifier.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    public void setDepartment(String department) {
+        this.department = department;
+    }
+
+    public String getSection() {
+        return section;
+    }
+
+    public void setSection(String section) {
+        this.section = section;
+    }
+
+
+    /**
+     * An object to hold identifiers composed according to pattern "{institution}.{faculty}.{department}.{section}".
+     *
+     * @param identifier A dot-separated string formatted according to pattern {integer}.{integer}.{integer}.{integer}
+     * @throws BadlyFormattedIdentifierException In case the identifier does not match the input pattern
+     */
+    public Identifier(String identifier) {
+        if (isNull(identifier) || identifier.isEmpty()) {
+            throw new BadlyFormattedIdentifierException(identifier);
+        }
+        if (identifier.split(MATCH_SEPARATOR).length == FOUR_SEGMENTS) {
+            String[] segments = identifier.split(MATCH_SEPARATOR);
+            this.institution  = segments[0];
+            this.faculty = segments[1];
+            this.department = segments[2];
+            this.section = segments[3];
+        } else {
+            throw new BadlyFormattedIdentifierException(identifier);
+        }
+    }
+
+    /**
+     * A method to create a top-level unit.
+     *
+     * @param institution An integer representing the top-level unit
+     */
+    public Identifier(int institution) {
+        this(String.format(IDENTIFIER_TEMPLATE,
+                institution, EMPTY_SUBUNIT, EMPTY_SUBUNIT, EMPTY_SUBUNIT));
+    }
+
+    /**
+     * A method to create a faculty level unit.
+     *
+     * @param institution An integer representing the top-level unit
+     * @param faculty     An integer representing the faculty level unit
+     */
+    public Identifier(int institution, int faculty) {
+        this(String.format(IDENTIFIER_TEMPLATE, institution, faculty, EMPTY_SUBUNIT, EMPTY_SUBUNIT));
+    }
+
+    /**
+     * A method to create a department level unit.
+     *
+     * @param institution An integer representing the top-level unit
+     * @param faculty     An integer representing the faculty level unit
+     * @param department  An integer representing the department-level unit
+     */
+    public Identifier(int institution, int faculty, int department) {
+        this(String.format(IDENTIFIER_TEMPLATE, institution, faculty, department, EMPTY_SUBUNIT));
+    }
+
+    /**
+     * A method to create a section level unit.
+     *
+     * @param institution An integer representing the top-level unit
+     * @param faculty     An integer representing the faculty level unit
+     * @param department  An integer representing the department-level unit
+     * @param section     An integer representing the section-level unit
+     */
+    public Identifier(int institution, int faculty, int department, int section) {
+        this(String.format(IDENTIFIER_TEMPLATE, institution, faculty, department, section));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: institution (declared at line 110), department (declared at line 110), faculty (declared at line 110), section (declared at line 110). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 111. Consider formatting this data with &#39;%d&#39; instead.